File: packages/ios/Sources/Components/DateRangePicker/DateRangePicker.figma.swift
import SwiftUI
import Figma

struct KozmosDateRangePickerConnect: FigmaConnect {
    let component = KozmosDateRangePicker.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=443-10939"

    @FigmaString("Label Text")
    var label: String = "Date range"

    @FigmaString("Start Label Text")
    var startLabel: String = "Start date"

    @FigmaString("End Label Text")
    var endLabel: String = "End date"

    @FigmaString("Start Value Text")
    var startValue: String = "May 21, 2026"

    @FigmaString("End Value Text")
    var endValue: String = "May 25, 2026"

    @FigmaString("Helper Text")
    var helperText: String = "Choose a date range."

    @FigmaBoolean("Show Helper Text")
    var showHelperText: Bool = false

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Focus": "focus",
            "Disabled": "disabled",
            "Readonly": "readonly"
        ]
    )
    var state: String = "default"

    @FigmaEnum(
        "Status",
        mapping: [
            "Default": KozmosInputStatus.`default`,
            "Error": KozmosInputStatus.error,
            "Warning": KozmosInputStatus.warning,
            "Success": KozmosInputStatus.success
        ]
    )
    var status: KozmosInputStatus = .default

    @FigmaEnum(
        "Content",
        mapping: [
            "Closed": false,
            "Open": true
        ]
    )
    var showsCalendarPreview: Bool = false

    var body: some View {
        KozmosDateRangePicker(
            startDate: .constant(figmaDate(self.startValue, fallback: "2026-05-21")),
            endDate: .constant(figmaDate(self.endValue, fallback: "2026-05-25")),
            label: self.label,
            startLabel: self.startLabel,
            endLabel: self.endLabel,
            helperText: self.showHelperText ? self.helperText : nil,
            disabled: self.state == "disabled",
            readOnly: self.state == "readonly",
            status: self.status,
            showsCalendarPreview: self.showsCalendarPreview
        )
    }
}

private func figmaDate(_ value: String, fallback: String) -> Date {
    for formatter in [figmaIsoDateFormatter, figmaDisplayDateFormatter] {
        if let date = formatter.date(from: value) {
            return date
        }
    }
    return figmaIsoDateFormatter.date(from: fallback) ?? Date()
}

private let figmaIsoDateFormatter: DateFormatter = {
    let formatter = DateFormatter()
    formatter.dateFormat = "yyyy-MM-dd"
    formatter.locale = Locale(identifier: "en_US_POSIX")
    formatter.timeZone = TimeZone(secondsFromGMT: 0)
    return formatter
}()

private let figmaDisplayDateFormatter: DateFormatter = {
    let formatter = DateFormatter()
    formatter.dateFormat = "MMM d, yyyy"
    formatter.locale = Locale(identifier: "en_US_POSIX")
    formatter.timeZone = TimeZone(secondsFromGMT: 0)
    return formatter
}()

File: packages/android/src/main/java/com/kozmos/components/DateRangePicker/DateRangePicker.figma.kt
package com.kozmos.components.daterangepicker

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.kozmos.components.input.KozmosInputStatus
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.util.Locale

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=443-10939")
class KozmosDateRangePickerConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Date range"

    @FigmaProperty(FigmaType.Text, "Start Label Text")
    val startLabel: String = "Start date"

    @FigmaProperty(FigmaType.Text, "End Label Text")
    val endLabel: String = "End date"

    @FigmaProperty(FigmaType.Text, "Start Value Text")
    val startValue: String = "May 21, 2026"

    @FigmaProperty(FigmaType.Text, "End Value Text")
    val endValue: String = "May 25, 2026"

    @FigmaProperty(FigmaType.Text, "Helper Text")
    val helperText: String = "Choose a date range."

    @FigmaProperty(FigmaType.Boolean, "Show Helper Text")
    val showHelperText: Boolean = false

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Focus" to "focus",
        "Disabled" to "disabled",
        "Readonly" to "readonly"
    )

    @FigmaProperty(FigmaType.Enum, "Status")
    val status: KozmosInputStatus = Figma.mapping(
        "Default" to KozmosInputStatus.Default,
        "Error" to KozmosInputStatus.Error,
        "Warning" to KozmosInputStatus.Warning,
        "Success" to KozmosInputStatus.Success
    )

    @FigmaProperty(FigmaType.Enum, "Content")
    val expanded: Boolean = Figma.mapping(
        "Closed" to false,
        "Open" to true
    )

    @Composable
    fun ComponentExample() {
        KozmosDateRangePicker(
            value = KozmosDateRangeValue(
                start = figmaDate(startValue, "2026-05-21"),
                end = figmaDate(endValue, "2026-05-25")
            ),
            onValueChange = {},
            label = label,
            startLabel = startLabel,
            endLabel = endLabel,
            helperText = if (showHelperText) helperText else null,
            enabled = state != "disabled",
            readOnly = state == "readonly",
            status = status,
            expanded = expanded
        )
    }
}

private val isoDateFormatter: DateTimeFormatter = DateTimeFormatter.ISO_LOCAL_DATE
private val displayDateFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("MMM d, yyyy", Locale.US)

private fun figmaDate(value: String, fallback: String): LocalDate =
    runCatching { LocalDate.parse(value, isoDateFormatter) }
        .getOrElse {
            runCatching { LocalDate.parse(value, displayDateFormatter) }
                .getOrElse { LocalDate.parse(fallback, isoDateFormatter) }
        }
